server/plugins/openAiDallE3Plugin.js

```diff
@@ -0,0 +1,105 @@
+import RequestDurationEstimator from '../../lib/requestDurationEstimator.js';
+import pubsub from '../pubsub.js';
+import ModelPlugin from './modelPlugin.js';
+
+const requestDurationEstimator = new RequestDurationEstimator(10);
+
+/**
+ * @description This plugin is for the OpenAI DALL-E 3 model.
+ */
+class OpenAIDallE3Plugin extends ModelPlugin {
+    constructor(config, pathway, modelName, model) {
+        super(config, pathway, modelName, model);
+    }
+
+    /**
+     * @description At the time of writing, the DALL-E 3 API on Azure is sync-only, so to support async
+     * we keep the request open and send progress updates to the client 
+     * over a websocket.
+     */
+    async execute(text, parameters, _, pathwayResolver) {
+        const url = this.requestUrl(text);
+        const data = JSON.stringify({ prompt: text });
+
+        let id;
+        const { requestId, pathway } = pathwayResolver;
+
+        const makeRequest = () => this.executeRequest(url, data, {}, { ...this.model.headers }, {}, requestId, pathway);
+
+        if (!parameters.async) {
+            // synchronous request
+            return await makeRequest();
+        }
+        else {
```

Comment:
How does the client get the id back in the async case so it knows where to subscribe for updates?